File: myscripts/python/testscripts/PiDigits.java
import java.io.*;
import java.net.*;

class PiDigits {
    public static void main(String argv[]) throws Exception {
        int i, j;
        int portno;
        int digits = 25000;
        int buffer_size = 1000;
        int total = 0;

        int SCALE = 10000;
        int ARRINIT = 2000;
        int[] arr = new int[digits+1];
        int carry = 0;

        for (i = 0; i <= digits; ++i) {
            arr[i] = ARRINIT;
        }

        for (i = digits; i > 0; i-=14) {
            int sum = 0;
            for (j = i; j > 0; --j) {
                sum = sum * j + SCALE * arr[j];
                arr[j] = sum % (j * 2 - 1);
                sum /= j * 2 - 1;
            }
            System.out.print (String.format("%04d", carry + sum / SCALE));
            carry = sum % SCALE;
        }

        if (argv.length > 0) {
        	try {
                        portno = Integer.parseInt(argv[0]);
	                Socket clientSocket = new Socket("localhost", portno);
	                byte[] buf = new byte[buffer_size];
	                DataOutputStream outToServer = new DataOutputStream(clientSocket.getOutputStream());
	                outToServer.write(buf);
	                System.out.println("Sent " + buffer_size + " bytes\n");
	                total += buffer_size;
	                clientSocket.close ();
			System.out.println("Sent " + total + " bytes\n");
                } catch (NumberFormatException e) {
                    System.err.println("Argument " + argv[1] + " must be an integer.");
                    System.exit(1);
                }
        }
    }
}

File: myscripts/python/testscripts/pidigits.py
import socket
import sys

total = 0
buffer_size = 1000

digits = 25000
scale = 10000
maxarr = 2000 + digits
arrinit = 2000
carry = 0
arr = [arrinit for i in range(digits+1)]

for i in xrange(digits, 1, -14):
    total = 0
    for j in xrange(i, 0, -1):
        total = (total * j) + (scale * arr[j])
        arr[j] = total % ((j * 2) - 1)
        total = total / ((j * 2) - 1)
    sys.stdout.write("%04d" % (carry + (total / scale)))
    carry = total % scale

if len (sys.argv) > 1:
    portno = int(sys.argv[1])
    # Create a TCP/IP socket
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

    # Connect the socket to the port where the server is listening
    server_address = ("localhost", portno)
    print >>sys.stderr, 'connecting to %s port %s' % server_address
    sock.connect(server_address)

    try:
        # Send data
        message = bytearray([1 for b in range(buffer_size) ])
        print >>sys.stderr, 'sending %d bytes' % len(message)
        sock.sendall(message)

        total += len(message)
        print >>sys.stderr, 'sent %d total bytes' % total

    finally:
        print >>sys.stderr, 'closing socket'
        sock.close()
